packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewManagersPropertyCache.java

```diff
@@ -393,6 +393,10 @@ public BoxedColorPropSetter(ReactProp prop, Method setter) {
    */
   /*package*/ static Map<String, PropSetter> getNativePropSettersForShadowNodeClass(
       Class<? extends ReactShadowNode> cls) {
+    if(cls == null) {
+      return new HashMap<>();
+    };
+
```

Comment:
Added null-check because `cls` was coming `null` when calling this function recursively with `cls.getSuperClass()` in [lines 410-413](https://github.com/facebook/react-native/pull/39005/files#diff-26375553a8545f236d50426a9d97c4933eff4e4addd7070166b4527e4dda9cc3R410-R413)

Reply:
Why is this change needed? Was one of your test failing?

Reply:
@cortinico Exactly, `testNativePropsIncludeCorrectTypes` was failing with a NullPointerException when instantiating the UIManagerModule:

```kotlin
val uiManagerModule = UIManagerModule(reactContext, viewManagers, 0)
```

I tracked it all the way down to ViewManagersPropertyCache.java, in `getNativePropSettersForShadowNodeClass`:

When this function is called recursively (in lines 410-413) with `cls.getSuperclass()`, the superclass might be null, causing `cls.getInterfaces()` (line 400) to crash.

```java
for (Class iface : cls.getInterfaces()) {
  ...
}
```

Let me know if you believe there is a better way to solve this!